timeline/views_ajax.py

```diff
@@ -211,4 +243,15 @@ def get(self, request, *args, **kwargs):
             'timeline.view_classified_site_event'
         ):
             return HttpResponseForbidden()
-        return Response(self.get_event_extra(event), status=200)
+
+        try:
+            if self.kwargs['idx']:
+                status = event.get_status_changes()
+                return Response(
+                    self.get_event_extra(
+                        event, status[int(self.kwargs['idx'])]
+                    ),
+                    status=200,
+                )
+        except KeyError:
+            return Response(self.get_event_extra(event), status=200)
```

Comment:
Here we also have quite a bit of repetition between the two views, please refactor.

Also I'm not sure if we only want to return non-status extra data when we encounter a `KeyError`?

Reply:
I am not sure, that refactoring this logic in a separatet Mixin is a good idea, as mostly this code is retriving GET. However, I optimized it a little bit. Maybe it is ok?